docs/source/about_dataset_load.rst

```diff
@@ -0,0 +1,101 @@
+Build and load
+==============
+
+Nearly every deep learning workflow begins with loading a dataset, which makes it one of the most important steps. With 🤗 Datasets, there are more than 900 datasets available to help you get started with your NLP task. All you have to do is call: :func:`datasets.load_dataset` to take your first step. This function is a true workhorse in every sense because it builds and loads every dataset you use.
+
+ELI5: ``load_dataset``
+-------------------------------
+
+Let's begin with a basic Explain Like I'm Five.
+
+At first, :func:`datasets.load_dataset` downloads and imports the dataset loading script associated with the requested dataset from the Hugging Face Hub. The Hub is a central repository where all the Hugging Face datasets and models are stored. Code in the loading script defines the dataset information (description, features, URL to the original files, etc.), and tells 🤗 Datasets how to generate and display examples from it.
```

Comment:
> downloads and imports the dataset loading script associated with the requested dataset from the Hugging Face Hub.

This is only true for Community datasets, isn't it? The loading script of canonical datasets is downloaded from GitHub.
Maybe @lhoestq can confirm this.

Reply:
All the datasets are listed on the Hub, but indeed the script of canonical datasets come from github.
Maybe if can be confusing to discuss this difference in details in the ELI5 part though

Reply:
Clarified that only community datasets downloads the script from the Hub, and canonical datasets downloads from Github. Also added to the `seealso` admonition a link to learn more about the differences between the two dataset types.